### PR TITLE
refactor assets to work with aws s3

### PR DESCRIPTION
**Pull request by spiridonovfed**
Issue #21

### CODE REVIEW

File: assets/.env [hunk 1]
```diff
@@ -14,8 +14,8 @@ POSTGRES_PORT=5432
 DATABASE_URL="postgresql+psycopg2://${POSTGRES_USER}:${POSTGRES_PASSWORD}@${POSTGRES_HOST}:${POSTGRES_PORT}/${POSTGRES_DB}"
 
 ENDPOINT=minio:9000
-MINIO_ACCESS_KEY=minioadmin
```
Comment (khyurri) [inline assets/.env:17]: Maybe better to use S3_ACCESS_KEY and S3_SECRET_KEY ?

File: assets/.env [hunk 2]
```diff
@@ -14,8 +14,8 @@ POSTGRES_PORT=5432
 DATABASE_URL="postgresql+psycopg2://${POSTGRES_USER}:${POSTGRES_PASSWORD}@${POSTGRES_HOST}:${POSTGRES_PORT}/${POSTGRES_DB}"
 
 ENDPOINT=minio:9000
```
Comment (khyurri) [inline assets/.env:16]: Maybe we can use S3_ENDPOINT?

File: assets/src/config.py [hunk 3]
```diff
@@ -29,8 +29,8 @@ class Config(BaseSettings):
     postgres_port: Optional[str]
     database_url: Optional[str]
     endpoint: Optional[str]
```
Comment (khyurri) [inline assets/src/config.py:31]: Also rename to `s3_endpoint`

File: assets/src/utils/minio_utils.py [hunk 4]
```diff
@@ -14,14 +15,21 @@
 
 MinioClient = minio.Minio(
     endpoint=settings.endpoint,
-    access_key=settings.minio_access_key,
-    secret_key=settings.minio_secret_key,
+    access_key=settings.MINIO_ROOT_USER,
+    secret_key=settings.MINIO_ROOT_PASSWORD,
+    secure=False,
```
Comment (khyurri) [inline assets/src/utils/minio_utils.py:20]: Create config variable `MINIO_SECURE_CONNECTION` and use this value from config

File: assets/src/utils/minio_utils.py [hunk 5]
```diff
@@ -14,14 +15,21 @@
...
+)
+
+
+MinioClientAWS = minio.Minio(
```
Comment (khyurri) [inline assets/src/utils/minio_utils.py:24]: You can check object creation arguments by checking if `settings.aws_profile_name` is not None?

**Review by khyurri — CHANGES_REQUESTED** [review]
Discussed changes in comments

File: assets/.env [hunk 6]
```diff
@@ -14,8 +14,8 @@ POSTGRES_PORT=5432
 DATABASE_URL="postgresql+psycopg2://${POSTGRES_USER}:${POSTGRES_PASSWORD}@${POSTGRES_HOST}:${POSTGRES_PORT}/${POSTGRES_DB}"
```
[the remark this answers is not in the document]
Comment (spiridonovfed) [reply assets/.env:15]: done, but should be tested on env - could be possible confilcts with "X-Current_Tenant" value in JWT

File: assets/.env (hunk 1 above)
Comment (spiridonovfed) [reply assets/.env:17]: MINIO_ROOT_USER and MINIO_ROOT_PASSWORD are used in .env for docker-compose in order to establish the default user in minio automatically for local usage.
Then, we use those variables to establish connection between our web_app and local minio.
If we work with AWS S3, these particular credentials are not used

File: assets/src/utils/minio_utils.py [hunk 7]
```diff
@@ -6,18 +6,31 @@
...
+    )
+else:
+    # Minio is used locally
+    minio_config.update(
+        {
+            "access_key": settings.minio_root_user,
+            "secret_key": settings.minio_root_password,
+        }
+    )
+MinioClient = minio.Minio(**minio_config)
```
Comment (spiridonovfed) [inline assets/src/utils/minio_utils.py:34]: Corrected - now we using only one MinioClient instance, but it's config varies

File: assets/.env (hunk 2 above)
Comment (spiridonovfed) [reply assets/.env]: agreed - renamed to s3_endpoint

File: assets/src/config.py (hunk 3 above)
Comment (spiridonovfed) [reply assets/src/config.py]: done

File: assets/src/utils/minio_utils.py [hunk 8]
```diff
@@ -14,14 +15,21 @@
...
-    access_key=settings.minio_access_key,
-    secret_key=settings.minio_secret_key,
+    access_key=settings.MINIO_ROOT_USER,
+    secret_key=settings.MINIO_ROOT_PASSWORD,
+    secure=False,
+)
+
+
+MinioClientAWS = minio.Minio(
+    endpoint="s3.amazonaws.com",
```
[the remark this answers is not in the document]
Comment (spiridonovfed) [reply assets/src/utils/minio_utils.py]: done

File: assets/src/utils/minio_utils.py (hunk 4 above)
Comment (spiridonovfed) [reply assets/src/utils/minio_utils.py]: done

File: assets/src/utils/minio_utils.py (hunk 5 above)
Comment (spiridonovfed) [reply assets/src/utils/minio_utils.py]: agreed - done

File: assets/src/utils/minio_utils.py (hunk 7 above)
Comment (spiridonovfed) [reply assets/src/utils/minio_utils.py:34]: done - using only one instance

File: assets/.env (hunk 1 above)
Comment (khyurri) [reply assets/.env:17]: You may setup `MINIO_ROOT_USER` and `MINIO_ROOT_PASSWORD` for local development using `environments` in `docker-compose.yaml` 

For application make sense to use `S3_ACCESS_KEY` and `S3_SECRET_KEY`

File: assets/.env (hunk 6 above)
[the remark this answers is not in the document]
Comment (khyurri) [reply assets/.env:15]: I cannot find S3_PREFIX variable in this config

File: assets/src/routers/minio_router.py [hunk 9]
```diff
@@ -155,13 +155,15 @@ async def create_bucket(
             HTTPException status 400
 
     """
+    if settings.bucket_prefix is not None:
+        bucket.name = f"{settings.bucket_prefix}-{bucket.name}"
```
Comment (khyurri) [inline assets/src/routers/minio_router.py:159]: Is it only one place where `bucket.name` is used?

Comment (spiridonovfed) [reply assets/src/routers/minio_router.py]: No, it's not.
I've added changes to other places, where tenant name from jwt is used as bucket.name.

File: assets/src/utils/minio_utils.py [hunk 10]
```diff
@@ -13,23 +13,28 @@
...
+    "endpoint": settings.s3_endpoint,
+    "secure": settings.minio_secure_connection,
+}
+if settings.aws_profile_name is not None:
+    # Minio is used with AWS S3
+    minio_config.update(
+        {"credentials": AWSConfigProvider(profile=settings.aws_profile_name)}
+    )
+else:
+    # Minio is used locally
```
[the remark this answers is not in the document]
Comment (spiridonovfed) [reply assets/src/utils/minio_utils.py]: Corrected

File: assets/.env (hunk 1 above)
Comment (spiridonovfed) [reply assets/.env:17]: I would like to show in code the usage of those variables and discuss it

**Review by khyurri — APPROVED** [review]
LGFM

**Pull request merged**